### PR TITLE
Mousquetaires additions

### PR DESCRIPTION
**Pull request by aleene**

### CODE REVIEW

File: taxonomies/ingredients.txt [hunk 1]
```diff
@@ -16873,7 +16880,7 @@ de:Rapsöl
 el:κραμβέλαιο
 es:aceite de colza, aceite de nabina
 fi:rapsiöljy, rypsiöljy
-fr:huile de colza, huile végétale de colza, huile colza, colza, graisses végétales de colza, graisses vegetales de colza
+fr:huile de colza, huile végétale de colza, huile colza, colza, graisses végétales de colza, graisses vegetales de colza, huilde de colza
```
Comment (teolemon) [inline taxonomies/ingredients.txt]: @aleene can you report the products with the typo, so that we can send them to Intermarché ?

Comment (stephanegigandet) [reply taxonomies/ingredients.txt]: We should not add typos in the taxonomy, it should only contain correct entries. We will use the taxonomy to do spellchecking, so it needs to contain only correct entries.

File: taxonomies/ingredients.txt [hunk 2]
```diff
@@ -265,7 +265,7 @@ el:Μέσο οξίνισης
 es:Acidulante, acidulant
 et:Hape
 fi:Happo
-fr:acidifiant, acide
+fr:acidifiant, acide, acidifant
```
Comment (stephanegigandet) [inline taxonomies/ingredients.txt]: acidifant is incorrect

File: taxonomies/ingredients.txt [hunk 3]
```diff
@@ -487,7 +487,7 @@ el:Γαλακτωματοποιητής
 es:Emulgente, Emulsificante
 et:Emulgaator
 fi:Emulgointiaine
-fr:Émulsifiant
+fr:Émulsifiant, émulsiﬁant
```
Comment (stephanegigandet) [inline taxonomies/ingredients.txt]: Misspelling, should not be added.

File: taxonomies/ingredients.txt [hunk 4]
```diff
@@ -10399,7 +10402,7 @@ vegetarian:en:yes
 <en:whey
 en:whey powder, powdered whey
 de:Molkenpulver
-fr:lactosérum en poudre, petit-lait en poudre, sérum de lait en poudre, poudre de lactosérum
+fr:lactosérum en poudre, petit-lait en poudre, sérum de lait en poudre, poudre de lactosérum, latosérum en poudre
```
Comment (stephanegigandet) [inline taxonomies/ingredients.txt]: Misspelling, should not be added.

File: taxonomies/ingredients.txt [hunk 5]
```diff
@@ -10614,6 +10617,10 @@ nl:volle room
 fr:crème pâtissière
 # ingredient/fr:crème-pâtissière has 105 products in french  @2019-02-03
 
+<en:cream
+en:wipped cream
```
Comment (stephanegigandet) [inline taxonomies/ingredients.txt]:
```suggestion
en:whipped cream
```

File: taxonomies/ingredients.txt [hunk 6]
```diff
@@ -24638,7 +24645,7 @@ es:jarabe de glucosa
 et:glükoosisiirup
 eu:Almibar
 fi:glukoosisiirappi
-fr:sirop de glucose
+fr:sirop de glucose, siropde glucose
```
Comment (stephanegigandet) [inline taxonomies/ingredients.txt]: Misspelling, should not be added.

File: taxonomies/ingredients.txt (hunk 3 above)
Comment (aleene) [reply taxonomies/ingredients.txt]: Is this a misspelling? They used a ﬁ, which is a single character.

**Comment by aleene** [comment]
OK I removed the misspellings. I note that the quality of mousquetaires is not very good.

File: taxonomies/ingredients.txt (hunk 3 above)
Comment (stephanegigandet) [reply taxonomies/ingredients.txt]: the "fi" character is not used in French, this one probably comes from an OCR engine

File: taxonomies/ingredients.txt [hunk 7]
```diff
@@ -16427,7 +16434,7 @@ de:pflanzliches Öl gehärtet
 <en:vegetable oil and fat
 en:palm oil and fat
 de:Palmöl und Palmfett, Palmöl und Fett
-fr:Huile et matière grasse de palme
+fr:Huile et matière grasse de palme, Matieres grasses et huile vegetales de palme
```
Comment (stephanegigandet) [inline taxonomies/ingredients.txt]:
```suggestion
fr:Huile et matière grasse de palme, Matières grasses et huile végétale de palme
```

File: taxonomies/ingredients.txt (hunk 1 above)
Comment (teolemon) [reply taxonomies/ingredients.txt]: Yes agreed, but we need to write those typo down

**Comment by teolemon** [comment]
```  
#   Failed test at t/additives.t line 502.
#     Structures begin differing at:
#          $got->[2] = 'en:cupric-sulfate'
#     $expected->[2] = 'en:copper-sulfate'
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
#   Failed test at t/additives.t line 527.
#     Structures begin differing at:
#          $got->[0] = 'en:calcium-carbonate'
#     $expected->[0] = 'en:mineral'
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
#   Failed test at t/additives.t line 571.
#     Structures begin differing at:
#          $got->[0] = 'en:calcium-carbonate'
#     $expected->[0] = 'en:mineral'
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
#   Failed test at t/additives.t line 607.
#     Structures begin differing at:
#          $got->[9] = 'en:cupric-sulfate'
#     $expected->[9] = 'en:copper-sulfate'
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# [
#   'en:e322i'
# ]
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# [
#   'en:e422',
#   'en:e331',
#   'en:e330',
#   'en:e500',
#   'en:e503'
# ]
# undef
#   Failed test at t/additives.t line 683.
#     Structures begin differing at:
#          $got->[0] = 'en:potassium'
#     $expected->[0] = 'en:mineral'
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
#   Failed test at t/additives.t line 761.
#     Structures begin differing at:
#          $got->[0] = 'en:calcium-phosphate'
#     $expected->[0] = 'en:mineral'
# [
#   'en:calcium-phosphate',
#   'en:magnesium-oxide',
#   'en:ferric-diphosphate',
#   'en:zinc-gluconate',
#   'en:cupric-gluconate',
#   'en:potassium-iodate',
#   'en:sodium-selenite'
# ]
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
#   Failed test at t/additives.t line 826.
#     Structures begin differing at:
#          $got->[0] = 'en:calcium'
#     $expected->[0] = 'en:mineral'
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
Use of uninitialized value $5 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $7 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $5 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $7 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $5 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $7 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $5 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $7 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $5 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
Use of uninitialized value $7 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1549.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
#   Failed test at t/additives.t line 950.
#     Structures begin differing at:
#          $got->[0] = 'en:calcium-citrate'
#     $expected->[0] = 'en:mineral'
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
#   Failed test at t/additives.t line 1008.
#     Structures begin differing at:
#          $got->[0] = 'en:calcium-phosphate'
#     $expected->[0] = 'en:mineral'
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
# undef
Use of uninitialized value $code in pattern match (m//) at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Products.pm line 159.
Use of uninitialized value $1 in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 1828.
Use of uninitialized value in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 2305.
spell correction (lc: fr): carbonates-dlammonium -> carbonates-d-ammonium - code: 
Use of uninitialized value in concatenation (.) or string at /home/travis/build/openfoodfacts/openfoodfacts-server/lib/ProductOpener/Ingredients.pm line 2305.
spell correction (lc: fr): tripoliphosphates -> tripolyphosphates - code: 
# undef
```

**Comment by teolemon** [comment]
poke @aleene @stephanegigandet

**Comment by stephanegigandet** [comment]
I'll try to fix the tests.

**Comment by stephanegigandet** [comment]
I made a mistake and commited the wrong file, I'll fix it.

**Pull request merged**